lib/Payplug/Authentication.php

```diff
@@ -144,4 +174,98 @@ private static function validateToken(Payplug $payplug)
             throw new ConfigurationException('The Payplug configuration requires a valid token.');
         }
     }
+
+    /**
+     * Retrieve client datas from the user manager resource.
+     *
+     * @param   Payplug $payplug the client configuration
+     *
+     * @return  array the client id and client_secret_mask
+     *
+     * @throws  Exception
+     */
+    public static function getClientData($session = null, Payplug $payplug = null)
+    {
+        if ($payplug === null) {
+            $payplug = Payplug::getDefaultConfiguration();
+        }
+        $kratosSession = self::setKratosSession($session);
+
+        $httpClient = new Core\HttpClient($payplug);
+        $response = $httpClient->get(Core\APIRoutes::$USER_MANAGER_RESOURCE, null, $kratosSession);
+        $result = array();
+        foreach ($response['httpResponse'] as $client) {
+            $result[] = array(
+                'client_id' => $client['client_id'],
+                'client_secret_mask' => $client['client_secret_mask'],
+                    'client_name' => $client['client_name'],
+                'client_type' => $client['client_type'],
+                'mode' => $client['mode'],
+
+            );
+        }
+
+        return $result;
+    }
+
+    /**
+     * Create a client ID and secret for a given mode
+     *
+     * @param $company_id
+     * @param $client_name
+     * @param $mode
+     * @param $session
+     * @param Payplug|null $payplug
+     * @return array
+     * @throws ConfigurationException
+     * @throws Exception\ConfigurationNotSetException
+     * @throws Exception\ConnectionException
+     * @throws Exception\HttpException
+     * @throws Exception\UnexpectedAPIResponseException
+     */
+    public static function createClientIdAndSecret($company_id='', $client_name='', $mode='', $session = null, Payplug $payplug = null)
```

Comment:
Same...

Reply:
i checked this branch on the plugin side: login, logout and simple payment